### ISSUE TITLE
Training is not saving accuracy values MaskRCNN

### ISSUE THREAD

**Issue by ameliafdez** [matterport/Mask_RCNN#2007]
hi! I'm trying to see accuracy while I'm training the model.

For that purpose, I have modified the model.py inside mrcnn folder like this:

![metrics](https://user-images.githubusercontent.com/50218100/74766878-251d6480-5286-11ea-995a-4432898380d3.PNG)

I have seen in so many forums that this works for so many people in other neural networks, but not for me in MaskRCNN. When I train again the model with this modification, is not showing accuracy and in history.history() is not saving it. Only the different losses defined in MaskRCNN. What I want to know is how to modify the implementation in order to obtain values 'acc' and 'val_acc' like this:

![example](https://user-images.githubusercontent.com/50218100/74766926-38303480-5286-11ea-9a73-4efbbc2de29c.PNG)

Thanks in advance!!

**Comment by suchiz** [comment]
Accuracy is something easy to compute for basic classifiers, because it's only based on the True positive, False positive, True negative, and False negative. But when it comes to further such as detection or segmentation, this metric isn't representative, and even in classifiers, accuracy isn't always the metrics that will lead to the best model. In networks suc has Mask-RCNN, we use mAP metric instead.

If you really need such a thing, take a look at:
- https://github.com/matterport/Mask_RCNN/issues/1839
- https://medium.com/@jonathan_hui/map-mean-average-precision-for-object-detection-45c121a31173
- And in ultils.py, see the compute_ap function. There, they calculate precision and recall metrics which are using TP, FP, FN... You would only need the last one which is TN, buuuut TN is not so obvious in MaskRCNN... Good luck to you

**Comment by ameliafdez** [comment]
very helpful answer @suchiz  !! thanks u so much

**Comment by filmonmehari** [comment]
Hi guys i am using the same model also, but i am in trouble of plotting the train_accuracy and validation_accuracy , moreover my advisor is asking me to show him the accuracy for my project but i couldn't find anny as far as my knowledge

Thanks in advance

**Comment by dingyan-dy** [comment]
> 

hello! Did you solve this issue? I also want to get the accuracy of train_dataset and val_dataset.
